oat-sa/tao-test-runner-qti-fe
# Fix/TR-702/Fix failing unit tests

Related to: https://oat-sa.atlassian.net/browse/TR-702

As the unit tests were failing silently on the CI, and loudly on local, several things have been made:
- Add a missing URI path in the environment config that was preventing the unit tests to complete
- Updated the package `tao-qunit-testrunner`
- Update several libraries
- Fixed several unit tests (list below)

Note: the unit tests may still fail sometimes due to a timeout on a particular module. Usually, starting the test again should work. This issue will need to be addressed in another story as it relates to the package `@oat-sa/tao-qunit-testrunner`

Updated unit tests:
- `test/plugins/content/dialog/itemInlineMessage/test.html`
- `test/plugins/content/responsiveness/collapser/test.html`
- `test/plugins/content/rubricBlock/test.html`
- `test/plugins/controls/duration/test.html`
- `test/plugins/controls/progressbar/plugin/test.html`
- `test/plugins/controls/timer/plugin/test.html`
- `test/plugins/controls/title/test.html`
- `test/plugins/navigation/allowSkipping/test.html`
- `test/plugins/navigation/next/linearNextItemWarning/test.html`
- `test/plugins/navigation/validateResponses/test.html`
- `test/plugins/tools/answerElimination/eliminator/test.html`
- `test/plugins/tools/answerElimination/plugin/test.html`
- `test/plugins/tools/answerMasking/plugin/test.html`
- `test/plugins/tools/apipTextToSpeech/plugin/test.html`
- `test/plugins/tools/areaMasking/plugin/test.html`
- `test/plugins/tools/highlighter/plugin/test.html`
- `test/plugins/tools/lineReader/plugin/test.html`
- `test/plugins/tools/magnifier/plugin/test.html`
- `test/proxy/qtiServiceProxy/test.html`
- `test/proxy/offline/proxy/test.html`


How to test:
- Run the following commands
```
npm i
npm run build
npm test
```
- All tests must pass (see the note above about possible random timeout)
- You may also start the local test server `npm run test:keepAlive`, then open your browser at the displayed URL, and access each test separately
